### PR TITLE
Add support for NumberBox

### PR DESCRIPTION
**Pull request by jeromelaban**
## PR Type

What kind of change does this PR introduce?
- Feature

## What is the new behavior?

Import implementation for the [`NumberBox`](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/number-box) control.

Release 2.3.191129002 (https://github.com/microsoft/microsoft-ui-xaml/commit/2237b09d9ba6052414ddccf3d61316de20564303)

~~This PR is in draft as there are some missing pieces, such as UI tests.~~

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current [supported SDKs](https://github.com/unoplatform/uno/blob/master/README.md#uno-features)
- [x] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [x] [Wasm UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Updated the [Release Notes](https://github.com/unoplatform/uno/tree/master/doc/ReleaseNotes)
- [x] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Comment by robloo** [comment]
This is amazing to see so quickly. Awesome work!

File: src/Uno.UI/Uno.UI.csproj [hunk 1]
```diff
@@ -362,6 +357,10 @@
 		</ProjectReference>
 	</ItemGroup>
 
+	<ItemGroup>
+	  <UpToDateCheckInput Remove="UI\Xaml\Controls\Primitives\RepeatButton.cs" />
+	</ItemGroup>
```
Comment (carldebilly) [inline src/Uno.UI/Uno.UI.csproj]: Remove!

File: src/Uno.UI/UI/Xaml/Controls/Primitives/RepeatButton/RepeatButton.cs [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+			ownerType: typeof(RepeatButton),
+			typeMetadata: new FrameworkPropertyMetadata(250, propertyChangedCallback: (s, e) => (s as RepeatButton)?.OnDelayChanged(e)));
+
+		public RepeatButton() : base()
+		{
+			InitializeVisualStates();
+
+			ClickMode = ClickMode.Press;
+
+			DefaultStyleKey = typeof(RepeatButton);
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/Controls/Primitives/RepeatButton/RepeatButton.cs]: 👍

File: src/Uno.UI.Tests/Windows_UI_Xaml_Data/xBindTests/Controls/Binding_Nullable.xaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+	  xmlns:local="using:Uno.UI.Tests.Windows_UI_Xaml_Data.xBindTests"
+	  xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
+	  xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
+	  mc:Ignorable="d">
+
+	<Grid>
+		<CheckBox x:Name="myCheckBox"
+				  x:FieldModifier="public" />
+		<TextBlock x:Name="_NullableBinding"
+				   x:FieldModifier="public"
```
[the remark this answers is not in the document]
Comment (jeromelaban) [reply src/Uno.UI.Tests/Windows_UI_Xaml_Data/xBindTests/Controls/Binding_Nullable.xaml]: Those are not used visually, so it's not a problem :)

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/NumberBox/NumberBox.Properties.cs [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+			DependencyProperty.Register("IsWrapEnabled", typeof(bool), typeof(NumberBox), new PropertyMetadata(false, (s, e) => (s as NumberBox)?.OnIsWrapEnabledPropertyChanged(e)));
+
+		public bool AcceptsExpression
+		{
+			get => (bool)GetValue(AcceptsExpressionProperty);
+			set => SetValue(AcceptsExpressionProperty, value);
+		}
+
+		public static readonly DependencyProperty AcceptsExpressionProperty =
+			DependencyProperty.Register("AcceptsExpression", typeof(bool), typeof(NumberBox), new PropertyMetadata(false /* ,UNO TODO (s, e) => (s as NumberBox)?.OnAcceptsExpressionPropertyChanged(e)*/));
```
[the remark this answers is not in the document]
Comment (jeromelaban) [reply src/Uno.UI/Microsoft/UI/Xaml/Controls/NumberBox/NumberBox.Properties.cs]: This is code that is not present in the original source, for some unknown reason.

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/NumberBox/NumberBox.cs [hunk 5]
```diff
@@ -0,0 +1,586 @@
...
+
+		TextBox m_textBox;
+		Windows.UI.Xaml.Controls.Primitives.Popup m_popup;
+
+		//RepeatButton.Click_revoker m_upButtonClickRevoker;
+		//RepeatButton.Click_revoker m_downButtonClickRevoker;
+		//TextBox.KeyDown_revoker m_textBoxKeyDownRevoker;
+		//TextBox.KeyUp_revoker m_textBoxKeyUpRevoker;
+		//RepeatButton.Click_revoker m_popupUpButtonClickRevoker;
+		//RepeatButton.Click_revoker m_popupDownButtonClickRevoker;
```
[the remark this answers is not in the document]
Comment (jeromelaban) [reply src/Uno.UI/Microsoft/UI/Xaml/Controls/NumberBox/NumberBox.cs]: Missing memory management, fixed :)

**Pull request merged**